### ISSUE TITLE
Enhance login call to also receive the fcm token

### ISSUE THREAD

**Issue by aniri** [code4romania/monitorizare-vot#247]
The endpoint `POST /api/v1/access/authorize` should also receive the fcm token needed for sending notifications. It should not be mandatory.

This should replace an extra call to `POST /api/v1/notification/register` after a successful login. 

Also, the fcm token could be used as unique device id and replace the `uniqueId` param.

**Comment by sandrohanea** [comment]
Hi!
I think I can make these changes.
Should we also change the endpoint to _api/v2/access_, in this case, right? Also, the _/api/v1/notification/register_ endpoint should be removed.

**Comment by aniri** [comment]
Hey @sandrohanea thanks for your interest in this project and your help! I failed to mention in the description that we should keep backwards compatibility. So we should enhance the v1 endpoint without affecting the previous functionality :/ 

Either by making a new v2 enpoint with the new functionality. And keep the old one as is. Or by using the uniqueId as the fcm token and add an optional channel param to the v1 request body.

**Comment by sandrohanea** [comment]
Hey @aniri,
Makes sense. I'll adapt my PR: https://github.com/code4romania/monitorizare-vot/pull/322 to keep the v1 endpoint.

I don't like the idea to reuse fields (uniqueId) as it will lead to workarounds(not store it if it's not an FCM token, etc) for older clients so I think the best approach, in this case, is to keep v1 as it is and add a new endpoint v2 with the desired functionality.

In this case, https://github.com/code4romania/monitorizare-vot-ong/pull/286 can be closed as the admin client won't be affected.

If this is ok, I can make these changes.

**Comment by aniri** [comment]
@sandrohanea yes, please! sounds like a good plan. sorry for not mentioning this in the initial description :(  you can also leave the pr in the admin web app. as eventually we will migrate to v2 and delete the old endpoint. but in the future :)

**Comment by sandrohanea** [comment]
Thanks @aniri 
I made the changes but I foresee a possible problem:
When this will be deployed to production and implemented on the mobile, some data migration is needed to clear MobileDeviceIds which contains the old unique ids otherwise the observers cannot log in to the system with a newer version of their app.

Do we want to address this issue in the scope of this PR(by checking if the stored MobileDeviceId in [Observers] table is v1 or v2 and allow them to use v2 even if v1 mobile device id is stored and doesn't match)?

**Comment by aniri** [comment]
@sandrohanea good call, but this won't be an issue as all the user accounts will be freshly added before using the apps. the only thing is that some of them will be using the older version of the apps (with v1) as only some of them update it (to v2).

**Comment by sandrohanea** [comment]
That shouldn't be a problem as long as they don't upgrade after the first login. 
If this happens we can either remove the existing code if in the db we have v1 and the user is authenticating with v2, either someone should manually reset the mobile id from the databse using the admin app.

**Comment by aniri** [comment]
Oh yes, you are right. I think removing the existing code if in the db we have v1 and the user is authenticating with v2 would be best in this case :)

**Comment by sandrohanea** [comment]
Hi @aniri,
I made the changes to keep v1 and to allow the user to log in when the "Lock device" feature is enabled and the user is authenticating with v2.
I tested those changes and they work as expected.

If the user is downgrading from v2 to v1 and the "Lock device" feature is enabled, he cannot log in but I think this is ok otherwise the whole "Lock Device" feature will be pointless(some user can log in with v1, other with v2, other with v1, etc).

PR: https://github.com/code4romania/monitorizare-vot/pull/322 is ready for review.
